ironfish/src/mining/pool.ts

```diff
@@ -183,15 +183,18 @@ export class MiningPool {
       return
     }
 
-    const blockTemplate = this.miningRequestBlocks.get(miningRequestId)
+    const originalBlockTemplate = this.miningRequestBlocks.get(miningRequestId)
 
-    if (!blockTemplate) {
+    if (!originalBlockTemplate) {
       this.logger.warn(
         `Client ${client.id} work for invalid mining request: ${miningRequestId}`,
       )
       return
     }
 
+    const blockTemplate = Object.assign({}, originalBlockTemplate)
```

Comment:
can't follow why you're doing this vs just using the original

Reply:
from the pr description: "When receiving work from miners, the pool will now clone the original block template instead of mutating it in place, which would cause issues later as it would try to re-broadcast for difficulty adjustment, etc"

tldr: is that we have to change the randomness on the block template before we try to serialize it and fail (which is the primary change in this PR: the try/catch). by changing it to a bad value, it's mutating the original block template which then later blows up when it tries to check for other things (ie, other users submitting work or block being re-broadcast with updated difficulty)

Reply:
OK - this is pretty subtle, ideally we would have a test that illustrates the issue. and breaks if we do changes the block in place.

I would be fine in this case with a ticket to come back for it, given there doesn't appear to be an obvious easy place to put such a test ATM.